src/components/CreateAssignment.js

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import Grid from '@mui/material/Grid';
+import Button from '@mui/material/Button';
+import Cookies from 'js-cookie';
+import {SERVER_URL} from '../constants.js';
+import DatePicker from "react-datepicker";
+
+import "react-datepicker/dist/react-datepicker.css";
+import {Link} from "react-router-dom";
+
+
+class CreateAssignment extends React.Component {
+    constructor(props) {
+        super(props);
+        console.log("CreateAssignment.cnstr "+ JSON.stringify(props.location));
+        this.state = { assignmentName: "", minDate: new Date().setHours(23,59,59,999), dueDate: new Date().setHours(0,0,0,0), selectedCourseId: new Number , courses: [] };
+    };
+
```

Comment:
CreateAssignment component looks good, looks like submit handles an assignment name with minimum of 6 characters, a due date, and course id. The fetch logic and return all look good.